File: sdkwork-ai-sdk-go/api/files.go
package api

import (
    "fmt"
    sdktypes "github.com/sdkwork/ai-sdk/types"
    sdkhttp "github.com/sdkwork/ai-sdk/http"
)

type FilesApi struct {
    client *sdkhttp.Client
}

func NewFilesApi(client *sdkhttp.Client) *FilesApi {
    return &FilesApi{client: client}
}

// List files
func (a *FilesApi) ListFiles(query map[string]interface{}) (sdktypes.OpenAiFileList, error) {
    raw, err := a.client.Get(AiApiPath("/files"), query, nil)
    if err != nil {
        var zero sdktypes.OpenAiFileList
        return zero, err
    }
    return decodeResult[sdktypes.OpenAiFileList](raw)
}

// Upload file
func (a *FilesApi) UploadFile(body *sdktypes.UploadFileRequest, query map[string]interface{}) (sdktypes.OpenAiFileObject, error) {
    raw, err := a.client.Post(AiApiPath("/files"), body, query, nil, "multipart/form-data")
    if err != nil {
        var zero sdktypes.OpenAiFileObject
        return zero, err
    }
    return decodeResult[sdktypes.OpenAiFileObject](raw)
}

// Retrieve file content
func (a *FilesApi) RetrieveFileContent(fileId string) (string, error) {
    raw, err := a.client.Get(AiApiPath(fmt.Sprintf("/files/%s/content", fileId)), nil, nil)
    if err != nil {
        var zero string
        return zero, err
    }
    return decodeResult[string](raw)
}

// Retrieve file
func (a *FilesApi) RetrieveFile(fileId string) (sdktypes.OpenAiFileObject, error) {
    raw, err := a.client.Get(AiApiPath(fmt.Sprintf("/files/%s", fileId)), nil, nil)
    if err != nil {
        var zero sdktypes.OpenAiFileObject
        return zero, err
    }
    return decodeResult[sdktypes.OpenAiFileObject](raw)
}

// Delete file
func (a *FilesApi) DeleteFile(fileId string) (sdktypes.OpenAiFileDeleteResponse, error) {
    raw, err := a.client.Delete(AiApiPath(fmt.Sprintf("/files/%s", fileId)), nil, nil)
    if err != nil {
        var zero sdktypes.OpenAiFileDeleteResponse
        return zero, err
    }
    return decodeResult[sdktypes.OpenAiFileDeleteResponse](raw)
}

File: sdkwork-ai-sdk-flutter/lib/src/api/files.dart
import '../http/client.dart';
import '../models.dart';

class FilesApi {
  final HttpClient _client;
  
  FilesApi(this._client);

  /// List files
  Future<OpenAiFileList?> listFiles(Map<String, dynamic>? params) async {
    final response = await _client.get(ApiPaths.aiPath('/files'), params: params);
    return response is OpenAiFileList ? response : null;
  }

  /// Upload file
  Future<OpenAiFileObject?> uploadFile(UploadFileRequest? body, Map<String, dynamic>? params) async {
    final response = await _client.post(ApiPaths.aiPath('/files'), body: body, params: params, contentType: 'multipart/form-data');
    return response is OpenAiFileObject ? response : null;
  }

  /// Retrieve file content
  Future<String?> retrieveFileContent(String fileId) async {
    final response = await _client.get(ApiPaths.aiPath('/files/${fileId}/content'));
    return response is String ? response : null;
  }

  /// Retrieve file
  Future<OpenAiFileObject?> retrieveFile(String fileId) async {
    final response = await _client.get(ApiPaths.aiPath('/files/${fileId}'));
    return response is OpenAiFileObject ? response : null;
  }

  /// Delete file
  Future<OpenAiFileDeleteResponse?> deleteFile(String fileId) async {
    final response = await _client.delete(ApiPaths.aiPath('/files/${fileId}'));
    return response is OpenAiFileDeleteResponse ? response : null;
  }
}
